application/src/main/kotlin/net/corda/v5/application/persistence/PersistenceService.kt

```diff
@@ -92,73 +92,26 @@ interface PersistenceService {
      * Find all entities of the same type from the persistence context in a single transaction.
      *
      * @param entityClass the type of the entities to find.
-     * @return list of entities found. Empty list if none were found.
+     * @return a [PagedQuery] that returns the list of entities found.
      * @throws CordaPersistenceException if an error happens during find operation
      */
     @Suspendable
-    fun <T : Any> findAll(entityClass: Class<T>): List<T>
+    fun <T : Any> findAll(entityClass: Class<T>): PagedQuery<T>
 
     /**
-     * Execute a named query in a single transaction. Casts results to the specified type [R].
+     * Execute a named query in a single transaction. Casts results to the specified type [T].
```

Comment:
Presumably this isn't executing the query any more, it's just generating it, and the user needs to fill in any parameters/offsets/limits before running it?

I wouldn't be against putting a quick usage example in here just to emphasise the point. (Our API docs are pretty bad in this regard, it's something we need to take a look at at some point soon.)

Reply:
I updated the kdocs and added a code example. I wasn't sure what the best way to do this is in Kdoc, I couldn't find an example, and I wasn't sure whether it should be Java or Kotlin, but have a look.
It seems like in Kdoc you can either do this by using markdown, or you can reference an actual sample, which is maybe something we should think about, but then we ought to do it for the entire API?

Reply:
ik

Reply:
Renders like this:
![image](https://user-images.githubusercontent.com/5557551/180163323-7daae432-479d-4034-8276-10dc31ae5c0b.png)

Reply:
I think the API looks really nice when you see it shown for real.

Reply:
Looks good! The whole API with examples thing is probably an epic in itself so let's not worry too much about that here. Something to think about though.

Reply:
> Is paging independent of the chunking of results to meet the avro message size limitation?

Yes. And in theory, once we have chunking on the queries, you wouldn't need pages as it will query for all results, then chunk them up, but that may not be great from a performance point of view, so it may actually be worth thinking about whether we should impose a limit, or at least warn people against it.